Release 1.10.0 [8116] (#106)
eProsima/Fast-RTPS-docs: docs/notes.rst

Version 1.10.0
==============

This release adds the following features:

* New built-in :ref:`comm-transports-shm`
* Transport API refactored to support locator iterators
* Added subscriber API to retrieve info of first non-taken sample
* Added parameters to fully avoid dynamic allocations
* History of built-in endpoints can be configured
* Bump to FastCDR v1.0.13.
* Bump to Fast-RTPS-Gen v1.0.4.
* Require CMake 3.5 but use policies from 3.13

It also includes the following bug fixes and improvements:

* Fixed alignment on parameter lists
* Fixed error sending more than 256 fragments.
* Fix handling of STRICT_REALTIME.
* Fixed submessage_size calculation on last data_frag.
* Solved an issue when recreating a publishing participant with the same GUID.

> * Solved issue where a publisher could block on write for a long time when, after a
>   large number of samples have been sent, a new subscriber is matched.

Reviewer: I took me 2-3 reads to understand the sentence, what about:
```suggestion
  * Solved issue where a publisher could block on write for a long time when a new subscriber (late joiner) is matched, if the publisher had already sent a large number of messages.
```

Author: 👍

* Correctly handling the case where lifespan expires at the same time on several samples.
* Solved some issues regarding liveliness on writers with no readers.
* Correctly removing changes from histories on keyed topics.
* Not reusing cache change when sample does not fit.
* Fixed custom wait_until methods when time is in the past.
* Several data races and ABBA locks fixed.
* Reduced CPU and memory usage.

> * Reduced flakyness of liveliness tests.

Reviewer: I believe is flakiness. Modify `docs/spelling_wordlist.txt` accordingly

Author: 👍
